$OpenBSD: patch-src_extension_implementation_script_cpp,v 1.2 2013/10/22 00:15:00 edd Exp $

Security fix for CVE-2012-6076 inkscape: Reads .eps files from /tmp instead of current working directory

Patch taken from Debian:
http://patch-tracker.debian.org/patch/series/view/inkscape/0.48.3.1-1.3/04-fix-LP911146.diff

--- src/extension/implementation/script.cpp.orig	Thu Dec 13 17:00:46 2012
+++ src/extension/implementation/script.cpp	Thu Oct 10 14:50:24 2013
@@ -956,7 +956,14 @@ int Script::execute (const std::list<std::string> &in_
     // assemble the rest of argv
     std::copy(in_params.begin(), in_params.end(), std::back_inserter(argv));
     if (!filein.empty()) {
-        argv.push_back(filein);
+        if(Glib::path_is_absolute(filein))
+            argv.push_back(filein);
+        else {
+            std::vector<std::string> buildargs;
+            buildargs.push_back(Glib::get_current_dir());
+            buildargs.push_back(filein);
+            argv.push_back(Glib::build_filename(buildargs));
+        }
     }
 
     int stdout_pipe, stderr_pipe;
